oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/DocumentStore.java

```diff
@@ -456,4 +456,18 @@ <T extends Document> T findAndUpdate(Collection<T> collection,
      */
     long determineServerTimeDifferenceMillis()
             throws UnsupportedOperationException, DocumentStoreException;
+
+    /**
+     * The maximum size a node name, in bytes. This is only a problem for long path.
+     */
+    int NODE_NAME_LIMIT = Integer.getInteger("oak.nodeNameLimit", 150);
+
+    /**
+     * Return the size limit for node name based on the document store implementation
+     *
+     * @return node name size limit
+     */
+    default int getSizeLimit() {
```

Comment:
I think this method should be renamed to better indicate where this limit applies. E.g. `getNodeNameLimit()`.